### ISSUE TITLE
Using `nextflow inspect` with Singularity

### ISSUE THREAD

**Issue by ewels** [nextflow-io/nextflow#4433]
The new `nextflow inspect` command is working nicely with Docker, but there is some odd behaviour for singularity.

### Requires wave, but shouldn't need to

```console
$ nextflow inspect -r 2.6.0 nf-core/viralrecon -profile singularity,test --outdir .
Inspect command with Singularity requires enabling 'wave' and 'freeze' settings
```

We don't need to _build_ the Singularity container here, it's already hardcoded in the script. We just want the URI. Is there any way that we can get to it? [Source here](https://github.com/nf-core/viralrecon/blob/3731dd3a32a67a2648ea22c2bd980c224abdaee2/modules/nf-core/bcftools/filter/main.nf#L6-L8).

```nextflow
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bcftools:1.16--hfe4b78e_1':
        'quay.io/biocontainers/bcftools:1.16--hfe4b78e_1' }"
```

### Crashes if wave installed but no singularity

When I try locally, I get a different error - probably related:

```console
$ nextflow inspect -r 2.6.0 nf-core/viralrecon -profile singularity,test --outdir .
ERROR ~ java.lang.IllegalStateException: Failed to pull singularity image
  command: singularity pull  --name depot.galaxyproject.org-singularity-ubuntu-20.04.img.pulling.1697732463429 https://depot.galaxyproject.org/singularity/ubuntu:20.04 > /dev/null
  status : 127
  message:
    bash: singularity: command not found
```

Here it's not trying to use Wave, but it is trying to do `singularity pull` on the image, when I'm only asking to return the URI.

**Comment by pditommaso** [comment]
Update to nextflow 23.10.0 !

**Comment by pvanheus** [comment]
This behaviour was seen with nextflow 23.10.0.5889.

**Comment by ewels** [comment]
Yup, my outputs above are all from 23.10.0

**Comment by muffato** [comment]
Hiya. I'm seeing the same problem with Nextflow 23.10.0-5889

I was following #4069 and I'm pretty sure I tested it at some point and it _was_ working with Singularity. So maybe a regression from when the integration with Wave was added ?

**Comment by pditommaso** [comment]
Ok, it is solved! f5bb829f
